File: ALL2/nginxApp/SimpleWebsite/cgi-bin/listUploads.pl
#!/usr/bin/perl
use strict;
use warnings;
use CGI qw(:standard);
use File::Basename;

# Dossier contenant les fichiers multimédias
my $dir = 'uploads/';
opendir(my $dh, $dir) or die "Impossible d'ouvrir le dossier '$dir': $!";

# Récupérer tous les fichiers du dossier
my @files = readdir($dh);
closedir($dh);

# Début du document HTML
print header('text/html; charset=UTF-8');
print start_html(-title => 'Galerie Multimédia', -charset => 'UTF-8');
print "<h1 style='text-align: center;'>Galerie Multimédia</h1>";
print "<div class='media-container' style='display: flex; flex-wrap: wrap; justify-content: center;'>";

# Traiter chaque fichier
foreach my $file (@files) {
    next if ($file =~ /^\.\.?$/); # Ignorer . et ..

    my $ext = lc((fileparse($file, qr/\.[^.]*/))[2]);

    print "<div class='media-item' style='margin: 10px; text-align: center;'>";

    if ($ext =~ /\.(jpg|jpeg|png|gif)$/) {
        print "<img src='$dir$file' style='max-width: 200px; max-height: 200px;' alt='$file'>";
    } elsif ($ext =~ /\.(mp4|webm)$/) {
        print "<video controls style='max-width: 200px;'>
                    <source src='$dir$file' type='video/$ext'>
                    Votre navigateur ne prend pas en charge la lecture de vidéos.
                </video>";
    } elsif ($ext =~ /\.(mp3|wav)$/) {
        print "<audio controls style='max-width: 200px;'>
                    <source src='$dir$file' type='audio/$ext'>
                    Votre navigateur ne prend pas en charge l'audio.
                </audio>";
    }

    print "<p>$file</p>";
    print "</div>";
}

print "</div>";
print end_html;

File: ALL2/nginxApp/SimpleWebsite/cgi-bin/listUploads.py
#!/usr/bin/python3
import os
import cgi
import cgitb
import mimetypes

# Activer le débogage en cas d'erreur
cgitb.enable()

# Dossier contenant les fichiers multimédias
dir_path = 'uploads/'

try:
    # Récupérer tous les fichiers du dossier
    files = os.listdir(dir_path)
except Exception as e:
    print("Content-Type: text/plain; charset=UTF-8")
    print()
    print(f"Impossible d'ouvrir le dossier '{dir_path}': {e}")
    exit()

# Début du document HTML
# print("Content-Type: text/html; charset=UTF-8")
# print()
print("""
<!DOCTYPE html>
<html lang="fr">
<head>
    <meta charset="UTF-8">
    <title>Galerie Multimédia</title>
    <style>
        .media-container {
            display: flex;
            flex-wrap: wrap;
            justify-content: center;
        }
        .media-item {
            margin: 10px;
            text-align: center;
        }
        img, video, audio {
            max-width: 200px;
            max-height: 200px;
        }
    </style>
</head>
<body>
<h1 style='text-align: center;'>Galerie Multimédia</h1>
<div class='media-container'>
""")

# Traiter chaque fichier
for file in files:
    if file.startswith('.'):
        continue  # Ignorer . et ..

    ext = os.path.splitext(file)[1].lower()

    print("<div class='media-item'>")

    if ext in ['.jpg', '.jpeg', '.png', '.gif']:
        print(f"<img src='{dir_path}{file}' alt='{file}'>")
    elif ext in ['.mp4', '.webm']:
        mime_type, _ = mimetypes.guess_type(file)
        print(f"""<video controls>
                    <source src='{dir_path}{file}' type='{mime_type}'>
                    Votre navigateur ne prend pas en charge la lecture de vidéos.
                </video>""")
    elif ext in ['.mp3', '.wav']:
        mime_type, _ = mimetypes.guess_type(file)
        print(f"""<audio controls>
                    <source src='{dir_path}{file}' type='{mime_type}'>
                    Votre navigateur ne prend pas en charge l'audio.
                </audio>""")
    print(f"<p>{file}</p>")
    print("</div>")

print("""
</div>
</body>
</html>
""")
